### PR TITLE
ci: replace deprecated `set-output` command with `GITHUB_OUTPUT` environment variable

### LINKED ISSUE #251: Fix set-output in Github Actions to be deprecated
Hi!

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
As per this URL, set-output of Github Actions is deprecated.

I would like to create a PR for a fix to this issue.
Thanks.

### PR DESCRIPTION
**Pull request by k1rnt**
resolved #251

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,10 +20,11 @@ jobs:
       - name: Check release version
         id: check-tag
         run: |
-          echo ::set-output name=version::$(echo ${{ github.event.ref }} | cut -d / -f 3 | cut -c2-)
+          echo version=$(echo ${{ github.event.ref }} | cut -d / -f 3 | cut -c2-) >> $GITHUB_OUTPUT
           if [[ ${{ github.event.ref }} =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-              echo ::set-output name=match::true
+              echo match=true >> $GITHUB_OUTPUT
           fi
+        shell: bash
```
Comment (k1rnt) [inline .github/workflows/build.yml:27]: If you want to run your platform on Windows with CI in the future
you will need to set up a shell.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -18,10 +18,11 @@ jobs:
       - name: Check release version
         id: check-tag
         run: |
-          echo ::set-output name=version::$(echo ${{ github.event.ref }} | cut -d / -f 3 | cut -c2-)
+          echo version=$(echo ${{ github.event.ref }} | cut -d / -f 3 | cut -c2-) >> $GITHUB_OUTPUT
           if [[ ${{ github.event.ref }} =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-              echo ::set-output name=match::true
+              echo match=true >> $GITHUB_OUTPUT
           fi
+        shell: bash
```
Comment (k1rnt) [inline .github/workflows/deploy.yml:25]: same as above

File: .github/workflows/build.yml (hunk 1 above)
Comment (lvaylet) [reply .github/workflows/build.yml:27]: Thanks @k1rnt for upgrading this code after GitHub's recent announcement.

Would you mind adding double quotes around `{environment_variable_name}={value}` (here and in `deploy.yaml` too) so the new code matches [GitHub's documentation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files)?

Then I am not sure about the Windows CI part. Let's not optimize prematurely. The CI works fine in its current state. I don't think we need the `shell` lines, at least in this PR. If you think it is a good practice, let's open a different issue and a different PR to discuss the matter. I like to keep concerns separated as much as possible.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (lvaylet) [reply .github/workflows/deploy.yml:25]: Same here.

**Review by lvaylet — CHANGES_REQUESTED** [review]
Thanks for this PR. I added a couple of request for changes.

File: .github/workflows/build.yml (hunk 1 above)
Comment (k1rnt) [reply .github/workflows/build.yml]: @lvaylet thanks for the request for change and suggestion for correction!

[fix: double quotes around](https://github.com/google/slo-generator/pull/252/commits/b1aea834b8bb7358c5e7ba1f15f6b8ec4cef209b)
[revert set shell](https://github.com/google/slo-generator/pull/252/commits/198706e632bd20a2fa0cfa98a52220e6bc7618ca)

I have corrected the points you pointed out in this commit.

The Windows CI was also not currently needed as you commented.
If you ever need it in the future, I'd be happy to help!

**Comment by k1rnt** [comment]
@lvaylet 
I've made corrections and left comments based on your comments!
Please re-review!

**Review by lvaylet — APPROVED** [review]
That's exactly what I had in mind. Thanks a lot for your this PR.

Note that I edited the PR title so it matches the [semantic commit messages](https://nitayneeman.com/posts/understanding-semantic-commit-messages-using-git-and-angular/#ci) convention we use on this repo.

**Pull request merged**